game/src/main/java/io/castles/game/Game.java

```diff
@@ -51,8 +51,9 @@ public Tile getTile(int x, int y) {
         return this.board.getTile(x, y);
     }
 
-    public Optional<Tile> getDrawnTile() {
-        return Optional.ofNullable(drawnTile);
+    public Tile getDrawnTile(Player player) {
+        this.validateAction(player, GameState.PLACE_TILE);
```

Comment:
the validation here makes the Optional<Tile> response obsolete as drawnTile is always set in PLACE_TILE state.

Reply:
That wasn't the sole purpose of the optional. When requesting the `GameDTO` while not being in `PLACE_TILE` state, we don't want an error to be raised, which would be the case right now. The drawn tile should be `null` or someting else in the `GameDTO`, if the player that is reloading the page didn't just draw a tile

Reply:
I'd consider your change a bug now. Why was it necessary in the first place?